src/test/unit/analysis/test_plugin_base.py

```diff
@@ -17,98 +12,70 @@
-        with pytest.raises(PluginInitException):
-            AnalysisBasePlugin(self)
-
-    @staticmethod
-    def test_start_stop_workers():
-        sleep(2)
-
-    def test_object_processing_no_children(self):
+@pytest.mark.cfg_defaults({
+    'dummy_plugin_for_testing_only': {
+        'threads': '2',
+    },
+    'expert-settings': {
+        'block-delay': '0.1',
+    }
+})
+@pytest.mark.AnalysisPluginClass.with_args(DummyPlugin)
+@pytest.mark.plugin_start_worker
+class TestPluginBaseCore:
+    def test_object_processing_no_children(self, analysis_plugin):
         root_object = FileObject(binary=b'root_file')
-        self.base_plugin.in_queue.put(root_object)
-        processed_object = self.base_plugin.out_queue.get()
-        self.assertEqual(processed_object.uid, root_object.uid, 'uid changed')
-        self.assertTrue('dummy_plugin_for_testing_only' in processed_object.processed_analysis, 'object not processed')
-        self.assertEqual(processed_object.processed_analysis['dummy_plugin_for_testing_only']['plugin_version'], '0.0', 'plugin version missing in results')
-        self.assertGreater(processed_object.processed_analysis['dummy_plugin_for_testing_only']['analysis_date'], 1, 'analysis date missing in results')
-
-    def test_object_processing_one_child(self):
+        analysis_plugin.in_queue.put(root_object)
+        processed_object = analysis_plugin.out_queue.get()
+        assert processed_object.uid == root_object.uid, 'uid changed'
+        assert 'dummy_plugin_for_testing_only' in processed_object.processed_analysis, 'object not processed'
+        assert processed_object.processed_analysis['dummy_plugin_for_testing_only']['plugin_version'] == '0.0', 'plugin version missing in results'
+        assert processed_object.processed_analysis['dummy_plugin_for_testing_only']['analysis_date'] > 1, 'analysis date missing in results'
+
+    def test_object_processing_one_child(self, analysis_plugin):
         root_object = FileObject(binary=b'root_file')
         child_object = FileObject(binary=b'first_child_object')
         root_object.add_included_file(child_object)
-        self.base_plugin.in_queue.put(root_object)
-        processed_object = self.base_plugin.out_queue.get()
-        self.assertEqual(processed_object.uid, root_object.uid, 'uid changed')
-        self.assertTrue(child_object.uid in root_object.files_included, 'child object not in processed file')
+        analysis_plugin.in_queue.put(root_object)
+        processed_object = analysis_plugin.out_queue.get()
+        assert processed_object.uid == root_object.uid, 'uid changed'
+        assert child_object.uid in root_object.files_included, 'child object not in processed file'
 
 
-class TestPluginBaseAddJob(TestPluginBase):
-
-    def test_analysis_depth_not_reached_yet(self):
+@pytest.mark.AnalysisPluginClass.with_args(DummyPlugin)
+class TestPluginBaseAddJob():
+    def test_analysis_depth_not_reached_yet(self, analysis_plugin):
```

Comment:
Tests like this one scream `pytest.mark.parametrize` but we don't have to do this in this PR

Reply:
Agree. I plan to cherry pick parts of #825 which also simplify testing code and use more fixtures.
After this we could possibly review all testcases and do such refactoring.